docs/DeveloperGuide.md

```diff
@@ -6,7 +6,134 @@
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+Our program is structured as pictured in the following class diagram:
+![ClassDiagram.png](diagrams/ClassDiagram.png)
+
+All the inheritors of Command are features that will be described below.
+
+### Addition feature
+
+The addition mechanism is facilitated by 'Parser', 'Add', 'ModuleList' and 'UI', where a Module object will be created according to the content inputted by the user and stored in the module list.
+
+Below is an example usage scenario and how the addition mechanism behaves at each step:
+
+Step 1: The user launches the application for the first time. A ModuleList object will be initiated with empty content.
+
+Step 2: The user input of "add m/cs2113 s/y2s1 mc/4 g/a" will be taken in for parser and an object of class 'Add' will be returned.
+
+Step 3: The execute method in the Add class that is overrides will be called with parameter of moduleList and module to be added passed. This will add the Module into the module list created in step 1.
+
+The sequence diagram below illustrates this add mechanism:
+
+![AddUML.png](diagrams/AddUML.png)
```

Comment:
Consider changing the following: 
1. The first arrowing pointing into the Duke object should point to the top of the activation bar (the Duke object should only be activiated when called with UserInput)
2. The same applies to `add(modToAdd)`, the arrow should point to the top of the activation bar
3. The acitivation bar for `Add` should last throughout the callback functions called (reference the Duke object to see its lifetime)